frontend/src/utilities/__tests__/useDraggableTable.spec.ts

```diff
@@ -0,0 +1,322 @@
+      currentTarget: {
+        id: 'item3',
+        classList: { add: jest.fn() },
+        setAttribute: jest.fn(),
+      },
+    } as unknown as React.DragEvent<HTMLTableRowElement>;
+
+    expect(renderResult.result.current.tableProps.tbodyProps.ref.current).toBe(ref.current);
+    expect(renderResult).hookToHaveUpdateCount(1);
+
+    act(() => {
+      onDragStart(dragStartEvent);
+    });
+
+    expect(renderResult).hookToHaveUpdateCount(2);
+    expect(dragStartEvent.dataTransfer.effectAllowed).toBe('move');
+    expect(dragStartEvent.dataTransfer.setData).toHaveBeenCalledWith('text/plain', 'item3');
+    expect(dragStartEvent.currentTarget.classList.add).toHaveBeenCalledWith('pf-m-ghost-row');
+    expect(dragStartEvent.currentTarget.setAttribute).toHaveBeenCalledWith('aria-pressed', 'true');
+    expect(renderResult.result.current.tableProps.className).toBe('pf-m-drag-over');
+
+    tableProps.tbodyProps.ref.current = null;
+  });
+});
+describe('onDrop', () => {
+  it('should handle onDrop behaviour with valid ref and event', () => {
+    const itemOrder = ['item3', 'item1', 'item2'];
+    const tableElement = generateRowsWithItems(itemOrder);
+    document.body.appendChild(tableElement);
+
+    const renderResult = testHook(useDraggableTable)(itemOrder, setItemOrder);
+
+    const { rowProps, tableProps } = renderResult.result.current;
+    const { onDrop } = rowProps;
+    const { ref } = tableProps.tbodyProps;
+    ref.current = document.body.querySelector('tbody');
+
+    if (ref.current) {
+      ref.current.getBoundingClientRect = jest.fn().mockReturnValue({
+        x: 40,
+        y: 40,
+        width: 20,
+        height: 20,
+      } as DOMRect);
+    }
+
+    act(() => {
+      onDrop({
+        clientX: 45,
+        clientY: 45,
+      } as unknown as React.DragEvent<HTMLTableRowElement>);
+    });
+
+    expect(renderResult).hookToHaveUpdateCount(1);
+    expect(setItemOrder).toHaveBeenCalledWith(itemOrder);
+    expect(renderResult.result.current.tableProps.tbodyProps.ref.current).toBe(ref.current);
+    expect(renderResult.result.current.tableProps.className).toBe(undefined);
+  });
+
+  it('should call onDragCancel when event is invalid and ref is null', () => {
+    const renderResult = testHook(useDraggableTable)(['item1', 'item2', 'item3'], setItemOrder);
+
+    const { rowProps, tableProps } = renderResult.result.current;
+    const { onDrop } = rowProps;
+    const { ref } = tableProps.tbodyProps;
+    ref.current = null;
+
+    expect(renderResult.result.current.tableProps.tbodyProps.ref.current).toBe(null);
+
+    act(() => {
+      onDrop({} as unknown as React.DragEvent<HTMLTableRowElement>);
+    });
+
+    expect(renderResult).hookToHaveUpdateCount(1);
+    expect(setItemOrder).not.toHaveBeenCalled();
+
+    tableProps.tbodyProps.ref.current = null;
+  });
+
+  it('should call onDragCancel when event is invalid but ref is not null', () => {
```

Comment:
`onDragCancel` results in new state as well as calls to the dom that don't get tested anywhere.